ui/app/AppLayouts/Chat/views/CommunityColumnView.qml

```diff
@@ -209,24 +209,27 @@ Item {
                 property var categoryItem
 
                 openHandler: function (id) {
-                    // Not Refactored Yet
-//                    categoryItem = root.store.chatsModelInst.communities.activeCommunity.getCommunityCategoryItemById(id)
+                    let jsonObj = root.communitySectionModule.getItemAsJson(id)
+                    let obj = JSON.parse(jsonObj)
+                    if (obj.error) {
+                        console.error("error parsing chat item json object, id: ", id, " error: ", obj.error)
+                        close()
+                        return
+                    }
+                    categoryItem = obj
                 }
 
                 StatusMenuItem {
-                    // Not Refactored Yet
                     enabled: communityData.amISectionAdmin
                     //% "Edit Category"
                     text: qsTrId("edit-category")
                     icon.name: "edit"
                     onTriggered: {
-                        // Not Refactored Yet
-//                        Global.openPopup(createCategoryPopup, {
-//                            communityId: root.store.chatsModelInst.communities.activeCommunity.id,
-//                            isEdit: true,
-//                            categoryId: categoryItem.id,
-//                            categoryName: categoryItem.name
-//                        })
+                       Global.openPopup(createCategoryPopup, {
+                           isEdit: true,
+                           categoryId: categoryItem.categoryId,
```

Comment:
This could be `categoryItem.id` instead, but since I did added `categoryId`, I guess it works as well.